types/reach__router/index.d.ts

```diff
@@ -177,6 +177,6 @@ export function useLocation(): WindowLocation;
 
 export function useNavigate(): NavigateFn;
 
-export function useParams(): any;
+export function useParams<TParams extends { [Param in keyof TParams]?: string } = {}>(): TParams;
```

Comment:
` = {}` is right, but also backwards-incompatible, as the default type of useParams becomes `{}` (strict) instead of `any`, therefore requiring to be explicitly cast before being used.

e.g.

    useParams().foo

would work before but not anymore.

I'm not sure what's the correct way to introduce a backwards-incompatible change in DT. It feels like  it should be ` = any` now and have a major version bump that changes this to `= {}`.

Reply:
Ahh I see, yeah that's a pickle. 🤔 I could leave a `TODO` about updating the `any` to `{}` in the next major release if that would be useful.

Reply:
Yeah that sounds good!